src/main/java/net/sf/jabref/gui/mergeentries/FetchAndMergeWorker.java

```diff
@@ -0,0 +1,84 @@
+package net.sf.jabref.gui.mergeentries;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.gui.BasePanel;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.IdBasedFetcher;
+import net.sf.jabref.logic.importer.fetcher.ArXiv;
+import net.sf.jabref.logic.importer.fetcher.DoiFetcher;
+import net.sf.jabref.logic.importer.fetcher.IsbnFetcher;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jdesktop.swingworker.SwingWorker;
+
+
+public class FetchAndMergeWorker extends SwingWorker<Optional<BibEntry>, Void> {
+
+    private static final Log LOGGER = LogFactory.getLog(FetchAndMergeWorker.class);
+
+    private final BasePanel panel;
+    private final BibEntry entry;
+    private final String field;
+    private final Optional<String> fieldContent;
+    private final String type;
+
+
+    public FetchAndMergeWorker(BasePanel panel, BibEntry entry, String field) {
+        this.panel = Objects.requireNonNull(panel);
+        this.entry = Objects.requireNonNull(entry);
+        this.field = Objects.requireNonNull(field);
+
+        this.fieldContent = entry.getField(field);
+        this.type = FieldName.getDisplayName(field);
+    }
+
+    @Override
+    protected Optional<BibEntry> doInBackground() throws Exception {
+        IdBasedFetcher fetcher;
```

Comment:
Please move this code to method `static Optional<IdBasedFetcher> getIdBasedFetcherForField(field, importFormatPreferences)` in new class `WebFetchers` (with s at the end)